token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -46,18 +45,22 @@ public DefaultOAuth2TokenKeyService(@Nonnull CloseableHttpClient httpClient) {
 	@Override
 	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId)
 			throws OAuth2ServiceException {
-		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, null);
+		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, null, null);
 	}
 
 	@Override
 	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId) throws OAuth2ServiceException {
+		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, clientId, null);
+	}
+
+	@Override
+	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId, @Nullable String azp) throws OAuth2ServiceException {
```

Comment:
refactor this method as well, same as above in the OAuth2TokenKeyServiceWithCache.java